src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -554,14 +554,46 @@ public virtual bool IsAnonymousType
         }
 
         /// <summary>
-        /// Is this a symbol for a Tuple
+        /// Is this a symbol for a Tuple.
         /// </summary>
         public virtual bool IsTupleType => false;
 
         /// <summary>
+        /// Verify if the given type can be used to back a tuple type and return cardinality of that tuple type in <paramref name="tupleCardinality"/>. 
+        /// </summary>
+        /// <param name="tupleCardinality">If method returns true, contains cardinality of the compatible tuple type.</param>
+        /// <returns></returns>
+        public virtual bool IsTupleCompatible(out int tupleCardinality)
+        {
+            tupleCardinality = 0;
+            return false;
+        }
+
+        /// <summary>
+        /// Verify if the given type can be used to back a tuple type. 
+        /// </summary>
+        public bool IsTupleCompatible()
```

Comment:
It is a very useful helper method. We are not done yet.

---
In reply to: [61820149](https://github.com/dotnet/roslyn/pull/11006#discussion_r61820149) [](ancestors = 61820149)